src/js/controllers/ctrl-selectedwidget.test.js

```diff
@@ -67,4 +69,35 @@ describe('adminWidgetController', () => {
 		$scope.hideModal.bind(this)()
 		expect(this.$parent.hideModal).toHaveBeenCalled()
 	})
+
+	it('does nothing if a widget is not editable', () => {
+		$scope.selected = {
+			editable: false
+		}
+
+		$scope.editWidget()
+
+		expect(Materia.Coms.Json.send).not.toHaveBeenCalled()
+	})
+
+	it('sets an alert message if the widget is locked already', () => {
+		$scope.selected = {
+			editable: true,
+			widget: {
+				id: 1
+			}
+		}
+
+		Namespace('Materia.Coms.Json').send = jest.fn().mockResolvedValueOnce({
+			is_locked: true,
+			can_publish: true
+		})
+
+		return $scope.jestTest._editWidgetPromise().then(() => {
+			expect(Materia.Coms.Json.send).toHaveBeenCalledWith('widget_instance_edit_perms_verify', [1])
+			expect($scope.alert.msg).toBe(
+				'This widget is currently locked, you will be able to edit this widget when it is no longer being edited by somebody else.'
```

Comment:
Update to match change above.
```suggestion
				'Someone else is editing this widget, you will be able to edit after they finish.'
```